bin/lean-gdb.py

```diff
@@ -1,3 +1,11 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+# Released under Apache 2.0 license as described in the file LICENSE.
+#
+# Author: Sebastian Ullrich
```

Comment:
> Ok, this only works if you run the lean binary in the bin directory, but not if you run the one in the cmake build directory.

Ah, I've never run that binary before :) .

> Sometimes, yes. But sometimes, I want to see the actual structure of an expression, and then you're completely lost, as the debugger will only show you the common fields for an expr since expr cells don't have RTTI info..

Well, you can already do `call print(e)` :) . I just looked up the hack for calling C++ functions from gdb, it's... beautiful http://stackoverflow.com/a/22798055/161659

> BTW, can you access the fields of the cell in a declaration from GDB? I always get "incomplete type".

Hm, I'm not sure I've ever tried that.